### PR TITLE
Clean up response security headers logic

### PR DESCRIPTION
**Pull request by claudijd**
Thanks @dylanwh for the suggestion!

### CODE REVIEW

File: scans/response.py [hunk 1]
```diff
@@ -10,19 +26,11 @@ def without_security_headers(self):
...
-            "X-Permitted-Cross-Domain-Policies": "none",
-            "X-XSS-Protection": "1; mode = block",
-            "Strict-Transport-Security": "max-age = 15768000; includeSubDomains"
-        }
+
+        # initialize headers if not already
+        if not security_response_dict.get("headers"):
+            security_response_dict["headers"] = {}
+
+        for header_name, header_value in self.SECURITY_HEADERS.items():
```
Comment (dylanwh) [inline scans/response.py:34]: You can use the dict constructor to perform this operation:
```python
>>> a = { 'prime': 5 }
>>> b = dict(a)
>>> b['prime'] = 7
>>> a
{'prime': 5}
>>> b
{'prime': 7}
```

File: scans/response.py [hunk 2]
```diff
@@ -10,19 +26,11 @@ def without_security_headers(self):
...
-            "X-Content-Type-Options": "nosniff",
-            "X-Download-Options": "noopen",
-            "X-Frame-Options": "DENY",
-            "X-Permitted-Cross-Domain-Policies": "none",
-            "X-XSS-Protection": "1; mode = block",
-            "Strict-Transport-Security": "max-age = 15768000; includeSubDomains"
-        }
+
+        # initialize headers if not already
+        if not security_response_dict.get("headers"):
```
Comment (dylanwh) [inline scans/response.py]: this should be 
```python
if "headers" not in security_response_dict:
```

File: scans/response.py (hunk 1 above)
Comment (claudijd) [reply scans/response.py:34]: I'm not sure I understand the feedback here.  Could you help me understand this better?

Comment (claudijd) [reply scans/response.py:34]: I'm merging this PR, but still interested in optimization opportunity here

**Pull request merged**